src/python/pants/base/exception_sink.py

```diff
@@ -418,7 +418,7 @@ def _exit_with_failure(cls, terminal_msg):
     cls._exiter.exit_and_fail(msg=formatted_terminal_msg, out=cls._interactive_output_stream)
 
   @classmethod
-  def _log_unhandled_exception_and_exit(cls, exc_class=None, exc=None, tb=None, add_newline=False):
+  def _log_unhandled_exception(cls, exc_class=None, exc=None, tb=None, add_newline=False):
```

Comment:
Looked up Typeshed for types on this one: https://github.com/python/typeshed/blob/75d9228b02c04a13b46c1d976f32859a88daff0c/stdlib/3/sys.pyi#L18-L19

```python
from typing import Optional, Type
from types import TracebackType

def _log_unhandled_exception(cls, 
  exc_class: Optional[Type[BaseException] = None, 
  exc: Optional[BaseException] = None, 
  tb: Optional[TracebackType] = None, 
  add_newline: bool = False,
) -> None:
  ...
```